### PR TITLE
(stablehlo.partition_id): add op handling and migrate to collectives.cc

### LINKED ISSUE #213: `stablehlo.partition_id`: return default id(0) on single-device.
### Operation name

stablehlo.partition_id: Returns the device ID. In single-device, this is `0`.

### Code to reproduce

```python
import jax
import jax.numpy as jnp
from jax import lax


def uses_axis_index(x):
    return x + lax.axis_index("i")

f = jax.pmap(uses_axis_index, axis_name="i")
x = jnp.array([7], dtype=jnp.int32)

print(f.lower(x).compiler_ir(dialect="stablehlo"))
```

```bash
module @jit__fun attributes {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
  func.func public @main(%arg0: tensor<1xi32> {mhlo.sharding = "{replicated}"}) -> (tensor<1xi32> {jax.result_info = "result", mhlo.sharding = "{replicated}"}) {
    %0 = call @shmap_body(%arg0) : (tensor<1xi32>) -> tensor<1xi32>
    return %0 : tensor<1xi32>
  }
  func.func private @shmap_body(%arg0: tensor<1xi32>) -> (tensor<1xi32> {jax.result_info = "[('i',)]"}) {
    %0 = stablehlo.reshape %arg0 : (tensor<1xi32>) -> tensor<i32>
    %c = stablehlo.constant dense<1> : tensor<ui32>
    %c_0 = stablehlo.constant dense<1> : tensor<ui32>
    %1 = stablehlo.partition_id : tensor<ui32>  # <--- No handler defined
    %2 = stablehlo.divide %1, %c : tensor<ui32>
    %3 = stablehlo.remainder %2, %c_0 : tensor<ui32>
    %4 = stablehlo.convert %3 : (tensor<ui32>) -> tensor<i32>
    %5 = stablehlo.add %0, %4 : tensor<i32>
    %6 = stablehlo.broadcast_in_dim %5, dims = [] : (tensor<i32>) -> tensor<1xi32>
    return %6 : tensor<1xi32>
  }
}
```

### Version and runtime information

```text
jax:     0.10.1
jaxlib:  0.10.1
jax-mps: 0.10.10
Python:  3.13.12
macOS:   26.5.1 (25F80)
Chip:    Apple M4
Memory:  16 GB
Backend: mps
```

### PR DESCRIPTION
**Pull request by MasterSkepticista**
This PR does two things:

* Adds a handler for `stablehlo.partition_id`. On single device, this is the device ID(0). Multi-partition modules are rejected. 
* Since this gets invoked in collective context, this op, alongwith `stablehlo.all_reduce` handling is migrated to `ops/collectives.cc`

Testing:
* Updated the unsupported-test skip pattern so `all_reduce` and `partition_id` are no longer treated as unsupported (unhandled, should I say) collectives.
* Added collectives test configs for single-device `lax.pmean` and `lax.axis_index`, and wired them into the main op test suite.
* Moved the existing all-reduce test out of reduction configs into the new collectives configs.

This will close #213.

### CODE REVIEW

File: scripts/_pytest_skip_unsupported_plugin.py [hunk 1]
```diff
@@ -72,8 +72,8 @@
     (
         re.compile(
             r"Unsupported operation\(s\): stablehlo\."
-            r"(all_reduce|all_gather|all_to_all|collective_permute|"
-            r"collective_broadcast|reduce_scatter|partition_id|replica_id)"
```
Comment (MasterSkepticista) [inline scripts/_pytest_skip_unsupported_plugin.py:76]: I could not get JAX to generate `stablehlo.replica_id` instruction.
It seems `replica_id` might be a legacy op name, with similar behavior as `partition_id`, hence removed.

File: src/pjrt_plugin/ops/collectives.cc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                       std::vector<mlx::core::array>& outputs, ExecContext& ctx) {
+    if (auto numPartitions = ctx.module->getAttrOfType<mlir::IntegerAttr>("mhlo.num_partitions")) {
+        if (numPartitions.getInt() != 1) {
+            ctx.error_message =
+                "stablehlo.partition_id: multi-partition execution is not supported";
+            MPS_LOG_ERROR("%s\n", ctx.error_message.c_str());
+            return false;
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (MasterSkepticista) [reply src/pjrt_plugin/ops/collectives.cc:66]: As I suspected, `stablehlo.replica_id` does not appear to be emitted by a normal user-facing JAX path in current JAX. The relevant public primitive is `lax.axis_index`, but its lowering uses `partition_id` whenever the axis context is a [`ShardingContext` or `SPMDAxisContext`](https://github.com/jax-ml/jax/blob/5a89435b6ad8392058a005da8a3c0ea1a8bb7099/jax/_src/lax/parallel.py#L2386). 

Current MLIR lowering defines `AxisContext` as only those two types in [mlir.py](https://github.com/jax-ml/jax/blob/5a89435b6ad8392058a005da8a3c0ea1a8bb7099/jax/_src/interpreters/mlir.py#L645), making the else: `hlo.replica_id()` branch effectively stale/unreachable for normal JAX lowering.

@tillahoffmann Do you suggest adding a handler anyway?

**Comment by tillahoffmann** [comment]
Thanks @MasterSkepticista, you were right to remove it; reinstating `replica_id` during the merge was my mistake. The skip list is only for things MPS fundamentally can't do (float64, multi-device collectives), not handlers we could implement. On a single device `replica_id` is just `0`, like `partition_id`, so I'd rather it fail loudly than get silently skipped.

Not worth a handler now: current jax only lowers `axis_index` to `partition_id`, so `replica_id` is unreachable and I couldn't test it. #221 drops it from the pattern.
